### PR TITLE
Add login component using tdd and login route

### PR DESCRIPTION
**Pull request by jason00111**

### CODE REVIEW

File: source/components/Login/Login.js [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import React from 'react'
+import Heading from '../reusable/Heading/Heading'
+
+const Login = ({ redirect }) => (
+  <div className='login-container'>
+    <Heading text='Step' />
+    <button className='login-button' onClick={ () => redirect( 'localhost:1337/auth/google' ) }>Login with Google</button>
```
Comment (nodatall) [inline source/components/Login/Login.js]: Pass the path down as a prop as well. 
```js
const Login = ({ redirect, path })
...
...
onClick={ () => redirect( path ) }

File: source/components/Login/LoginContainer.js [hunk 2]
```diff
@@ -0,0 +1,12 @@
+import React, { Component } from 'react'
+import Login from './Login'
+
+export default class LoginContainer extends Component {
+  static redirect(path) {
+    location.href = path
+  }
+
+  render() {
+    return <Login redirect={ LoginContainer.redirect } />
```
Comment (nodatall) [inline source/components/Login/LoginContainer.js]: This does not appear to be working. Undefined is being passed to the child, not the intended redirect function.

File: source/components/Login/_tests/Login.test.js [hunk 3]
```diff
@@ -0,0 +1,34 @@
+import React from 'react'
+import { mount } from 'enzyme'
+import sinon from 'sinon'
+import { expect } from '../../../../configuration/testSetup'
+import LoginContainer from '../LoginContainer'
+
+describe.only( '<LoginContainer />', () => {
```
Comment (nodatall) [inline source/components/Login/_tests/Login.test.js]: Take out .only

File: source/components/Login/_tests/Login.test.js [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+
+  const spy = sinon.spy(LoginContainer, 'redirect')
```
Comment (nodatall) [inline source/components/Login/_tests/Login.test.js]: Paren spaces plox

File: source/components/Login/_tests/Login.test.js [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+
+  it( 'should render login component with class \'login-container\'', () =>
+    expect( wrapper.find( '.login-container' ).length ).to.equal( 1 )
+  )
+
+  it( 'should render button with class \'login-button\'', () =>
+    expect( wrapper.find( '.login-button' ).length ).to.equal( 1 )
+  )
+
+  it( 'should go to route /auth/google when button clicked', () => {
```
Comment (nodatall) [inline source/components/Login/_tests/Login.test.js]: I am not sure why this is passing if it is not working in the browser. 
Investigate further, or switch to a non-static method. There is a part in this React docs that talks about static methods: 
https://facebook.github.io/react/docs/higher-order-components.html

File: source/components/Login/_tests/Login.test.js [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+
+    wrapper.find('.login-button').simulate('click')
+
+    expect(spy.called).to.equal(true)
+    expect(spy.calledWith('localhost:1337/auth/google')).to.equal(true)
```
Comment (nodatall) [inline source/components/Login/_tests/Login.test.js]: Paren spaces everywheree!!!!

File: source/components/Login/_tests/Login.test.js (hunk 5 above)
Comment (jason00111) [inline source/components/Login/_tests/Login.test.js]: We discovered that there was a mistake in the react routing.  We believe this test is still valid.

**Pull request merged**